src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,173 @@
+        assertEquals(LOGIN_TOO_SHORT, exception.getMessage());
+
+        user.setLogin("1234");
+        exception = assertThrows(InvalidUserException.class,
+                () -> registrationService.register(user));
+        assertEquals(LOGIN_TOO_SHORT, exception.getMessage());
+
+        user.setLogin("12345");
+        exception = assertThrows(InvalidUserException.class,
+                () -> registrationService.register(user));
+        assertEquals(LOGIN_TOO_SHORT, exception.getMessage());
+    }
+
+    @Test
+    void register_Login_Ok() {
+        User user = new User();
+        user.setLogin("123456");
+        user.setPassword("123456");
+        user.setAge(18);
+        assertDoesNotThrow(() -> registrationService.register(user));
+        assertEquals(user, storageDao.get(user.getLogin()));
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
+        User user = new User();
+        user.setLogin("123456");
+        user.setPassword("12345");
+        InvalidUserException exception = assertThrows(InvalidUserException.class,
+                () -> registrationService.register(user));
+        assertEquals(PASSWORD_TOO_SHORT, exception.getMessage());
+
+        user.setPassword("1");
+        exception = assertThrows(InvalidUserException.class,
+                () -> registrationService.register(user));
+        assertEquals(PASSWORD_TOO_SHORT, exception.getMessage());
+
+        user.setPassword("12");
+        exception = assertThrows(InvalidUserException.class,
+                () -> registrationService.register(user));
+        assertEquals(PASSWORD_TOO_SHORT, exception.getMessage());
+        user.setPassword("123");
+        exception = assertThrows(InvalidUserException.class,
+                () -> registrationService.register(user));
+        assertEquals(PASSWORD_TOO_SHORT, exception.getMessage());
+        user.setPassword("1234");
+        exception = assertThrows(InvalidUserException.class,
+                () -> registrationService.register(user));
+        assertEquals(PASSWORD_TOO_SHORT, exception.getMessage());
+        user.setPassword("12345");
+        exception = assertThrows(InvalidUserException.class,
+                () -> registrationService.register(user));
+        assertEquals(PASSWORD_TOO_SHORT, exception.getMessage());
+    }
+
+    @Test
+    void register_invalidAge_notOk() {
+        User user = new User();
+        user.setLogin("123456");
+        user.setPassword("123456");
+        user.setAge(16);
+        InvalidUserException exception = assertThrows(InvalidUserException.class,
+                () -> registrationService.register(user));
+        assertEquals(AGE_TOO_YOUNG, exception.getMessage());
+
+        user.setAge(-16);
+        InvalidUserException exceptionNegativeAge = assertThrows(InvalidUserException.class,
+                () -> registrationService.register(user));
+        assertEquals(AGE_TOO_YOUNG, exception.getMessage());
```

Comment:
The variable `exceptionNegativeAge` is declared but not used. You should use this variable in the `assertEquals` statement instead of `exception` to ensure the correct exception is being tested.